code/modules/antagonists/horror/horror_datums.dm

```diff
@@ -0,0 +1,710 @@
+					L.grippedby(user, instant = TRUE)
+					L.Knockdown(30)
+				return
+			if(INTENT_DISARM)
+				if(iscarbon(L))
+					var/mob/living/carbon/C = L
+					var/obj/item/I = C.get_active_held_item()
+					if(I)
+						if(C.dropItemToGround(I))
+							playsound(loc, "sound/weapons/whipgrab.ogg", 30)
+							target.visible_message("<span class='danger'>[I] is whipped out of [C]'s hand by [user]!</span>","<span class='userdanger'>A tentacle whips [I] out of your hand!</span>")
+							return
+						else
+							to_chat(user, "<span class='danger'>You can't seem to pry [I] off [C]'s hands!</span>")
+							return
+					else
+						C.attack_hand(user)
+						return
+	. = ..()
+
+/obj/item/horrortentacle/afterattack(atom/target, mob/user, proximity)
+	if(isliving(user.pulling) && user.pulling != target)
+		var/mob/living/H = user.pulling
+		user.visible_message("<span class='warning'>[user] throws [H] with [user.p_their()] [src]!</span>", "<span class='warning'>You throw [H] with [src].</span>")
+		H.throw_at(target, 8, 2)
+		H.Knockdown(30)
+		return
+	if(!proximity)
+		return
+	if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = target
+
+		if((!A.requiresID() || A.allowed(user)) && A.hasPower())
+			return
+		if(A.locked)
+			to_chat(user, "<span class='warning'>The airlock's bolts prevent it from being forced!</span>")
+			return
+
+		if(A.hasPower())
+			user.visible_message("<span class='warning'>[user] jams [src] into the airlock and starts prying it open!</span>", "<span class='warning'>You start forcing the airlock open.</span>",
+			"<span class='italics'>You hear a metal screeching sound.</span>")
+			playsound(A, 'sound/machines/airlock_alien_prying.ogg', 150, 1)
+			if(!do_after(user, 150, target = A)) //tentacle 50% worse than an armblade
+				return
+		user.visible_message("<span class='warning'>[user] forces the airlock to open with [user.p_their()] [src]!</span>", "<span class='warning'>You force the airlock to open.</span>",
+		"<span class='italics'>You hear a metal screeching sound.</span>")
+		A.open(2)
+		return
+	. = ..()
+
+/obj/item/horrortentacle/suicide_act(mob/user) //funnily enough, this will never be called, since horror stops suicide
+	user.visible_message("<span class='suicide'>[src] coils itself around [user] tightly gripping [user.p_their()] neck! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	return (OXYLOSS)
+
+//Pinpointer
+/obj/screen/alert/status_effect/agent_pinpointer/horror
+	name = "Soul locator"
+	desc = "Find your target soul."
+
+/datum/status_effect/agent_pinpointer/horror
+	id = "horror_pinpointer"
+	minimum_range = 0
+	range_fuzz_factor = 0
+	tick_interval = 20
+	alert_type = /obj/screen/alert/status_effect/agent_pinpointer/horror
+
+/datum/status_effect/agent_pinpointer/horror/scan_for_target()
+	return
+
+/datum/status_effect/agent_pinpointer/horror/point_to_target()
+	if(!ishorror(owner))
+		return
+	return ..()
+
+
+
+//ABILITIES
+

```

Comment:
Didn't look at any of below code, first glance seems meh, someone else needs to take a look